swarm/src/handler.rs

```diff
@@ -603,3 +685,5 @@ impl Ord for KeepAlive {
         }
     }
 }
+
+static EMPTY_HASHSET: Lazy<HashSet<String>> = Lazy::new(HashSet::new);
```

Comment:
As far as I can tell `HashSet::new` does not allocate:

> The hash set is initially created with a capacity of 0, so it will not allocate until it is first inserted into.

https://doc.rust-lang.org/std/collections/struct.HashSet.html#method.new

With that in mind, I don't think the use of `once_cell::Lazy` provides any value, correct?

Reply:
Great! I should have checked that.

Reply:
So actually, the reason I am doing this is not allocation but because the borrow-checker otherwise complains:

```
error[E0515]: cannot return value referencing temporary value
   --> swarm/src/handler.rs:300:9
    |
300 | /         ProtocolsAdded {
301 | |             protocols: protocols.difference(&HashSet::new()).peekable(),
    | |                                              -------------- temporary value created here
302 | |         }
    | |_________^ returns a value referencing data owned by the current function
```